src/app-logic/__mocks__/l10n.js

```diff
@@ -0,0 +1,18 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+const realL10n = jest.requireActual('../l10n.js');
+
+// This mocks en-US only.
+function mockFetchMessages(_locale: string) {
+  // Always return the english locale, whatever the requested locale
+  const fs = require('fs').promises;
+  return fs.readFile('../../../locales/en-US/app.ftl', 'UTF-8');
```

Comment:
I think the return pattern is not similar to the `fetchMessages` function, I tried `retun ['en-US', fs.readFile('../../../locales/en-US/app.ftl', 'UTF-8')];` this didn't resolve the problem in the tests. The test error's then was this, 
![image](https://user-images.githubusercontent.com/37832457/109017120-13fd0400-76e1-11eb-9e69-3537a80040eb.png)

Reply:
ah yeah I forgot about this.

You can try this:
*  add `async` for the function fetchMessages, remove the `_` for the parameter `_locale`
* use this code then inside:
```js
const localeContent = fs.readFile('../../../locales/en-US/app.ftl', 'UTF-8');
return [locale, localeContent];
```

The problem is:
* the function's return value should be `Promise<[localeString, localeContent]>`.
* But in my code I was returning `Promise<localeContent>` (the function `fs.readFile` returns a promise).
* when you tried `return ['en-US', fs.readFile('../../../locales/en-US/app.ftl', 'UTF-8')]`, you returned `[localeString, Promise<localeContent>]`

By making the function `async`, we make it return a Promise always. And we can `await` the result of `readFile` easily.

To compare, and so that you understand the difference, without the `async` version, this is what we should have done:
```js
return fs.readFile(...).then(content => [locale, content]);
```
This is exactly equivalent to the version with an async function. But the use of `await` makes it easier to follow, by making it more linear.